Change scope of RRD pages

willmcb opened the pull request:
- makes the Pending tab also have 'destroy' cases that are in the future. Those that have passed their destruction dates and have a destroy status move to the 'Ready for removal' tab. 
- makes the Ready for removal page only have cases where the destruction
  date is >= Date.today (i.e. no future dates)
- cases that are marked as 'Destroy' will remain on Pending tab until their planned destruction date is passed at which point they pass to ready for removal tab.


## Self-review checklist
<!-- Action these things before requesting reviews -->
* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
[CT-4220](https://dsdmoj.atlassian.net/browse/CT-4220)
[CT-4221](https://dsdmoj.atlassian.net/browse/CT-4221)

### Manual testing instructions
- change some cases destruction dates to fit inside and outside the rules specified above, they should be in the right places at the right times 👍

Commit:
Change scope of RRD pages

- makes the Pending tab also have 'destroy' cases
- makes the Ready for removal page only have cases where the destuction
  date is >= Date.today (i.e. no future dates)
- the range for when cases get to the tabs is any time before today + 4
  months

zheileman reviewed (approved):
👍 
I will update the filters in one of my next PRs, as I removed the "deleted" state from the pending to delete but now we can make use of it.

Commit:
Use aasm constants to refer to states

Also extact some date ranges into the RetentionSchedule class so they
are less 'magical'
